shared/src/api/extension/api/windows.ts

```diff
@@ -81,62 +100,79 @@ class ExtWindow implements sourcegraph.Window {
         }
     }
 
+    /**
+     * Perform a delta update (update/add/delete) of this window's view components.
+     */
+    public _update(data: WindowData): void {
+        const key = (c: ViewComponentData<Pick<sourcegraph.TextDocument, 'uri'>>): string => `${c.type}:${c.item.uri}`
+
+        const seenKeys = new Set<string>()
+        for (const c of data.visibleViewComponents) {
+            const k = key(c)
```

Comment:
I found this code hard to read with all the single letter var names.

```suggestion
            const key = buildKey(c)
```